pylint/config/callback_actions.py

```diff
@@ -38,3 +47,261 @@ def __call__(
         option_string: Optional[str] = None,
     ) -> None:
         return None
+
+
+class _AccessRunObjectAction(_CallbackAction):
+    """Action that has access to the Run object."""
+
+    def __init__(
+        self,
+        option_strings: Sequence[str],
+        dest: str,
+        nargs: None = None,
+        const: None = None,
+        default: None = None,
+        arg_type: None = None,
+        choices: None = None,
+        required: bool = False,
+        help_msg: str = "",
+        metavar: str = "",
+        **kwargs: "Run",
+    ) -> None:
+        self.run = kwargs["Run"]
+
+        super().__init__(
+            option_strings,
+            dest,
+            0,
+            const,
+            default,
+            arg_type,
+            choices,
+            required,
+            help_msg,
+            metavar,
+        )
+
+    @abc.abstractmethod
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Union[str, Sequence[Any], None],
+        option_string: Optional[str] = None,
+    ) -> None:
+        raise NotImplementedError
+
+
+class _MessageHelpAction(_CallbackAction):
+    """Display the help message of a message."""
+
+    def __init__(
+        self,
+        option_strings: Sequence[str],
+        dest: str,
+        nargs: None = None,
+        const: None = None,
+        default: None = None,
+        arg_type: None = None,
+        choices: None = None,
+        required: bool = False,
+        help_msg: str = "",
+        metavar: str = "",
+        **kwargs: "Run",
+    ) -> None:
+        self.run = kwargs["Run"]
+        super().__init__(
+            option_strings,
+            dest,
+            "+",
```

Comment:
`--help-msg` was the only callback that uses arguments.

However, to be able to work with `argparse` we do need to change one thing. We can no longer allow:
``pylint --help-msg=missing-docstring``
But only:
``pylint --help-msg missing-docstring``.

This is because callable actions cannot also store a value like the `=` operator does. I have changed the tests accordingly.

Reply:
That's a small breaking change, we might want to put this in the changelog but I guess no one is using the help message in a script, so...

Reply:
Yeah I think this would be fine without one. But let me know if you want me to add one!